File: 242-valid-anagram/valid-anagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
    if(s.length !== t.length) return false;

    const sFreqArr = new Array(26).fill(0);

    for(let i = 0; i < s.length; i++){
        sFreqArr[s[i].charCodeAt(0) - 'a'.charCodeAt(0)]++;
        sFreqArr[t[i].charCodeAt(0) - 'a'.charCodeAt(0)]--;
    }

    // for(let i = 0; i < t.length; i++){
    //     sFreqArr[t[i].charCodeAt(0) - 'a'.charCodeAt(0)]--;
    // }

    for(const freq of sFreqArr){
        if(freq !== 0) return false;
    }

    return true;
};

File: 242-valid-anagram/valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if(s.length() != t.length()){
            return false;
        }

        int n = s.length();

        int[] freq = new int[26];

        for(int i = 0; i < n; i++){
            freq[s.charAt(i) - 'a']++;
            freq[t.charAt(i) - 'a']--;
        }

        for(int num: freq){
            if(num != 0){
                return false;
            }
        }

        return true;
    }
}